src/lib/zf/zf_stackdump.c

```diff
@@ -26,6 +26,41 @@ static void dump_zockets(SkewPointer<zf_stack> stack, Zocket* zockets,
   }
 }
 
+void dump_attributes(SkewPointer<zf_stack_impl> stimpl)
+{
+  zf_dump("---------------------attributes-------------------------------\n");
+  if(stimpl->sti_tx_ring_max==-1){
+  zf_dump("tx_ring size= 512\n");
+  }
+  else{
+    zf_dump("tx_ring size= %d\n", stimpl->sti_tx_ring_max);
+  }
+  if(stimpl->sti_rx_ring_max==-1){
+  zf_dump("rx_ring size= 512\n");
+  }
+  else{
+    zf_dump("rx_ring size= %d\n", stimpl->sti_rx_ring_max);
+  }
+  zf_dump("tx_timestamping= %d\n", stimpl->sti_tx_timestamping);  
+  zf_dump("rx_timestamping= %d\n", stimpl->sti_rx_timestamping);  
+  zf_dump("ctpio= %d\n", stimpl->sti_ctpio);
+  zf_dump("ctpio_mode= %s\n", stimpl->sti_ctpio_mode);
+  zf_dump("pio= %d\n", stimpl->sti_pio);
+  zf_dump("reactor_spin_count= %d\n", stimpl->sti_reactor_spin_count);
+  zf_dump("tcp_timewait_wait= %d\n", stimpl->sti_tcp_timewait_ms);  
+  zf_dump("alt_buf_size= %d\n", stimpl->sti_alt_buf_size);  
+  zf_dump("alt_count= %d\n", stimpl->sti_alt_count);  
+  zf_dump("rx_ring_refill_batch_size= %d\n", stimpl->sti_rx_ring_refill_batch_size);
+  zf_dump("tcp_alt_ack_rewind= %d\n", stimpl->sti_tcp_alt_ack_rewind);  
+  zf_dump("tcp_delayed_ack= %d\n", stimpl->sti_tcp_delayed_ack);  
+  zf_dump("tcp_finwait_ms= %d\n", stimpl->sti_tcp_finwait_ms);  
+  zf_dump("tcp_wait_for_time_ms= %d\n", stimpl->sti_tcp_wait_for_time_wait);  
+  zf_dump("ctpio_max_frame_len= %d\n", stimpl->sti_ctpio_max_frame_len);
+  zf_dump("force_separate_tx_vi= %d\n", stimpl->sti_force_separate_tx_vi);
+  zf_dump("rx_ring_refill_interval= %d\n", stimpl->sti_rx_ring_refill_interval);
+  zf_dump("udp_ttl= %d\n", stimpl->sti_udp_ttl); 
```

Comment:
I think this value can become stale, as it's [used in zfut_alloc](https://github.com/Xilinx-CNS/tcpdirect/blob/85cf55a048ea2ed25dd91b8e3dbcc6ad9b8638f1/src/lib/zf/udp_tx.c#L88) to set the TTL on a per-socket basis, so I don't think we should be printing it at the stack level. It would be worth a check over the code to see where these other attributes are used to make sure these other ones can't be changed in any meaningful way to this stack after it's allocated.

Reply:
Ah, yes, I see what you mean. Perhaps in addition to having it logged here, it would be worth extending [`dump_zockets`](https://github.com/Xilinx-CNS/tcpdirect/blob/52de0c78ee45361c51b45fc5728e420b0c3fdf13/src/lib/zf/zf_stackdump.c#L30) to output each zockets view of this value, in addition to the initial value of ttl set at stack allocation time.

Reply:
@KalepuAMD I agree with Josh it would be good to add it to the dumping of the zocket data. 
If you trace it through you can dump it in the following function:
[https://github.com/Xilinx-CNS/tcpdirect/blob/52de0c78ee45361c51b45fc5728e420b0c3fdf13/src/lib/zf/udp_tx.c#L120](https://github.com/Xilinx-CNS/tcpdirect/blob/52de0c78ee45361c51b45fc5728e420b0c3fdf13/src/lib/zf/udp_tx.c#L120)

Reply:
added the ip->tll to the zocket information in UDP TX zockets. 

```
dellr250v:~/devel/fgit/test-onload-git/tcpdirect/build/gnu_x86_64/bin$ ./zf_stackdump dump
============================================================
onload version=95264a9bfe 2024-06-13 onload-8.1
tcpdirect version=8.1.3 onload-8.1 tcpdirect-8.1 215a4ac 2024-05-29 ON-15038
name=ens1f0/10a interface=ens1f0 vlan_id=65535
  pool: pkt_bufs_n=17536 free=17024
  config: tcp_timewait_ticks=666 tcp_finwait_ticks=666 ctpio_threshold=-1
  config: tcp_initial_cwnd=0 ms_per_tcp_tick=90
  alts: n_alts=0
  stats: ring_refill_nomem=0 cplane_alien_ifindex=0
         tcp_retransmits=0
  discards: discard_csum_bad=0 discard_mcast_mismatch=0
         discard_crc_bad=0 discard_trunc=0 discard_rights=0
         discard_ev_error=0 discard_other=0 discard_inner_csum_bad=0
         non_tcpudp=0
  hwport_to_nicno: [0,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,]
nic0: vi=266 vi_flags=1000000 nic_flags=1 intf=ens1f0 index=14 hw=1C1
  txq: pio_buf_size=4096 added=622350 removed=622349
============================================================
UDP TX ens1f0/10a:0 lcl=192.168.130.233:9001 rmt=192.168.130.232:9001 ttl=64
  path: dst=192.168.130.232 src=192.168.130.233 nic=0
   L2: 14: 00 0f 53 71 37 20 00 0f 53 7c 20 e0 08 00
   L3: 20: 45 00 00 28 00 00 40 00 40 11 b3 a2 c0 a8 82 e9 c0 a8 82 e8
   L4:  8: 23 29 23 29 00 14 c0 2e
------------------------------------------------------------
---------------------attributes-------------------------------
tx_ring size=512
rx_ring size=512
tx_timestamping=0
rx_timestamping=0
ctpio=1
ctpio_mode=sf-np
pio=3
reactor_spin_count=128
tcp_timewait_wait=60000
alt_buf_size=40960
alt_count=0
rx_ring_refill_batch_size=16
tcp_alt_ack_rewind=65536
tcp_delayed_ack=1
tcp_finwait_ms=60000
tcp_wait_for_time_ms=0
ctpio_max_frame_len=-1
force_separate_tx_vi=0
rx_ring_refill_interval=1
udp_ttl=64
log_level=11111111
```